src/tests/ftest/slurm_setup.py

```diff
@@ -60,17 +64,35 @@ def update_config_cmdlist(args):
     # Copy the slurm*example.conf files to /etc/slurm/
     if execute_cluster_cmds(all_nodes, COPY_LIST, args.sudo) > 0:
         sys.exit(1)
-
-    cmd_list = [
-        "sed -i -e 's/ControlMachine=linux0/ControlMachine={}/g' {}".format(
-            args.control, SLURM_CONF),
-        "sed -i -e 's/ClusterName=linux/ClusterName=ci_cluster/g' {}".format(
-            SLURM_CONF),
-        "sed -i -e 's/SlurmUser=slurm/SlurmUser={}/g' {}".format(
-            args.user, SLURM_CONF),
-        "sed -i -e 's/NodeName/#NodeName/g' {}".format(
-            SLURM_CONF),
-        ]
+    # grep SLURM_CONF to determine format of the the file
+    command = r"grep SlurmctldHost {}".format(SLURM_CONF)
+    task = run_task(all_nodes, command)
+    # Create a dictionary of hosts for each unique return code
+    results = {code: hosts for code, hosts in task.iter_retcodes()}
```

Comment:
(pylint-unnecessary-comprehension) Unnecessary use of a comprehension, use dict(task.iter_retcodes()) instead.